File: psuedocode/activities/problem_2/find_coding_region.cc
#include <iostream>
#include <set>
#include <string>

using namespace std;

string find_coding_region(string const& s){

  int i = 0;
  set<string> S = {"TAA", "TAG", "TGA"};

  while (i +2 <= s.length()){
      if (s.substr(i, 3) == "ATG"){
        int j = i + 3;
        while (j + 2 <= s.length()){
          if (S.find(s.substr(j, 3)) != S.end()){
            return s.substr(i + 3, j -(i + 3));
          }
          j += 3;
        }
      }
      i++;
  }
  return "";
}

int main(){
  string s;
  while (cin >> s){
    cout << find_coding_region(s) << endl;
   }
  return 0;
}

File: psuedocode/activities/problem_2/find_coding_region.py
import sys

def find_coding_region(s):
    i = 0
    while i + 2 <= len(s):
        if s[i:i+3] == "ATG":
            j = i + 3
            while j + 2 <= len(s):
                if s[j: j+3] in {"TAA", "TAG", "TGA"}:
                    return s[i+3 : j]
                j += 3
        i += 1 
    return ""


s = sys.stdin.readline()

while s != "":
    print(find_coding_region(s))
    s = sys.stdin.readline()
